ocw/lib/azure.py

```diff
@@ -124,33 +137,31 @@ def get_keeping_image_names(self, bbsrv):
 
     def cleanup_all(self):
         ''' Cleanup all autodateed data which might created during automated tests.'''
-        block_blob_service = BlockBlobService(account_name='openqa', account_key=self.get_storage_key())
-
-        self.cleanup_bootdiagnostics(block_blob_service)
+        self.cleanup_bootdiagnostics()
 
-        keep_images = self.get_keeping_image_names(block_blob_service)
-        self.cleanup_sle_images_container(block_blob_service, keep_images)
+        keep_images = self.get_keeping_image_names()
+        self.cleanup_sle_images_container(keep_images)
         self.cleanup_disks_from_rg(keep_images)
         self.cleanup_images_from_rg(keep_images)
         for i in keep_images:
             logger.info("Keep image {} ".format(i))
 
-    def cleanup_bootdiagnostics(self, block_blob_service):
-        containers = block_blob_service.list_containers()
+    def cleanup_bootdiagnostics(self):
+        containers = self.bs_client().list_containers()
         for c in containers:
             logger.debug('Found container {}'.format(c.name))
             if (re.match('^bootdiagnostics-', c.name)):
-                self.cleanup_bootdiagnostics_container(block_blob_service, c)
-
-    def cleanup_bootdiagnostics_container(self, bbsrv, container):
-        last_modified = container.properties.last_modified
-        generator = bbsrv.list_blobs(container.name)
-        for blob in generator:
-            if (last_modified < blob.properties.last_modified):
```

Comment:
you changed the logic from comparing the latest modified file inside that containter to the first modificated file (like a creation date for that containter.

Not sure if we really want this. I thought, if there is a bootdiagnostics containter where someone continously write to it, I wouldn't delete it.

Reply:
yeah this line , bring some doubts to me . but I come up with this change during fixing unit tests :) it is hard to discuss in such github format also I need some time to refresh full picture . So I heard you :)  but we need to discuss this when you will be back ...